bson/src/main/org/bson/json/JsonObject.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.json;
+
+import org.bson.BsonDocument;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+
+/**
+ * A wrapper class that holds a JSON object string. This class makes decoding JSON efficient.
+ * Note that this class only holds valid JSON objects, not arrays or other values.
+ *
+ * @since 4.2
+ */
+public class JsonObject implements Bson {
+    private final String json;
+
+    /**
+     * Constructs a new instance with the given JSON object string. Clients must ensure
+     * they only pass in valid JSON objects to this constructor. The constructor does not
+     * perform full validation on construction, but an invalid JsonObject can cause errors
+     * when it is used later on.
+     *
+     * @param json the JSON object string
+     */
+    public JsonObject(final String json) {
+        if (json == null) {
+            throw new IllegalArgumentException("Json cannot be null");
+        }
+
+        for (int i = 0; i < json.length(); i++) {
```

Comment:
would `new JsonObject("")` pass this test?

Could you use Assertions.isTrue for this check?

Reply:
Nice catch, fixed and added a test case. I used `Assertions.isTrueArgument` instead because we're just sanity checking the argument so I figured we would want an `IllegalArgumentException` instead.

Reply:
:+1: